generics/services/project.js

```diff
@@ -67,6 +64,171 @@ const templateLists = function (userToken, externalId) {
   });
 };
 
+/**
+ * Fetches the program Details based on the given Id.
+ * @param {string} userToken - The user's authentication token.
+ * @param {string[]|string} programId - ProgramId
+ * @returns {Promise<Object>} A promise that resolves to an object indicating success and containing the fetched data if successful.
+ */
+
+const programDetails = function (userToken, programId) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      // Construct the URL for the project service
+      let url = `${projectServiceUrl}${process.env.PROJECT_SERVICE_NAME}${messageConstants.endpoints.PROGRAM_DETAILS}/${programId}`;
+      // Set the options for the HTTP GET request
+      const options = {
+        headers: {
+          'content-type': 'application/json',
+          'X-auth-token': userToken,
+        },
+      };
+
+      request.get(url, options, projectServiceCallback);
+      let result = {
+        success: true,
+      };
+      // Handle callback fucntion
+      function projectServiceCallback(err, data) {
+        if (err) {
+          result.success = false;
+        } else {
+          let response = data.body;
+          let result = JSON.parse(response);
+          if (result.status === httpStatusCode['ok'].status) {
+            return resolve(result);
+          } else {
+            result.success = false;
+          }
+        }
+        return resolve(result);
+      }
+      setTimeout(function () {
+        return resolve(
+          (result = {
+            success: false,
+          })
+        );
+      }, messageConstants.common.SERVER_TIME_OUT);
+    } catch (error) {
+      return reject(error);
+    }
+  });
+};
+
+/**
+ * update the program  based on the given Id.
+ * @param {string} userToken - The user's authentication token.
+ * @param {string[]|string} programId - ProgramId
+ * @param {object} reqBody - update query
+ * @returns {Promise<Object>} update success message
+ */
+const programUpdate = function (userToken, programId, reqBody) {
```

Comment:
@praveenKDass please provide information on who and in which workflow will the program need to be updated?